src/hooks/__tests__/useWebSocket.test.ts

```diff
@@ -0,0 +1,106 @@
+import WebSocketServer from 'jest-websocket-mock';
```

Comment:
By using `jest-websocket-mock` we can have a local WebSocket server, that acts as a stub for receiving and sending messages. This allows us to test the entire flow, similar to how it'd look like using a real WebSocket server.